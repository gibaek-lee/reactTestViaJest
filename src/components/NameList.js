import React, { Component } from 'react'
import PropTypes from 'prop-types';

class NameList extends Component {
  static propTypes = {
    nameList: PropTypes.array
  }
  static defaultProps = {
    nameList: []
  }
  render() {
    const { nameList } = this.props
    const namesPackage = nameList.map(
      (name, i)=>(
        <li key={i}>{name}</li>
      )
    );
    return (
      <div>
        {namesPackage}
      </div>
    );
  }
}

export default NameList
