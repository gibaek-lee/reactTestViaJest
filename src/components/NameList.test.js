import React from 'react';
import { shallow } from 'enzyme';
import NameList from './NameList'

describe('NameList', () => {
  let component = null;

  /* component initial render test */
  it('renders correctly', () => {
    /* render에 영향을 미치는 부모컴포넌트로 부터 전달되는 변수(NameList)는
    초기값으로 예상되는 값 ['긴냥이', '이기백']을 직접 써준다. */
    component = shallow(<NameList nameList={['긴냥이', '이기백']} />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  /* redux + react: props.nameList [] 전달 가정한 rendering test*/
  it('is 긴냥이, 이기백', () => {
    const list = component.find('li');
    expect(list.at(0).text()).toBe('긴냥이');
    expect(list.at(1).text()).toBe('이기백');
  });
})
