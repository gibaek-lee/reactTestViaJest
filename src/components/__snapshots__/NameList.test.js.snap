// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList matches snapshot 1`] = `
<div>
  <li
    key="0"
  >
    긴냥이
  </li>
  <li
    key="1"
  >
    이기백
  </li>
</div>
`;
